## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :x: [more...](#5353b93a-b41b-4628-bc8c-4bcf772e8900) |
| mint-large-bucket.sh | :x: [more...](#b1cd6b10-f8aa-4bce-bfbc-23817eee356c) |
| mint-fs.sh | :x: [more...](#d5f4b67b-7edc-4ef4-bfe3-75375a88650a) |
| mint-dist-xl.sh | :x: [more...](#7857e461-3e83-4bd4-bb9c-940252e66f48) |
| mint-gateway-s3.sh | :x: [more...](#a5b24040-fdff-4334-8d03-6e6e7524e3fe) |
| mint-gateway-azure.sh | :x: [more...](#ab872c68-9359-4b26-96d8-4ebc96741930) |
| mint-zoned.sh | :x: [more...](#7e4fe1ba-8efa-4c67-9d71-70c161838903) |
| mint-gateway-nas.sh | :x: [more...](#5d3a8aaa-f619-4543-b9d6-bb747f44fda6) |
<a name="b1cd6b10-f8aa-4bce-bfbc-23817eee356c"></a>
### 9957-a4426e9/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:30304
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 536dbaeb8c46:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 5 minutes and 6 seconds
(2/15) Running aws-sdk-java tests ... FAILED in 2 seconds
Jul 02, 2020 5:04:42 AM com.amazonaws.util.Base64 warn
WARNING: JAXB is unavailable. Will fallback to SDK implementation which may be less performant.If you are using Java 9+, you will need to include javax.xml.bind:jaxb-api as a dependency.
com.amazonaws.services.s3.model.AmazonS3Exception: Server not initialized, please try again. (Service: Amazon S3; Status Code: 503; Error Code: XMinioServerNotInitialized; Request ID: 161DD70A689B8323; S3 Extended Request ID: 2dfe849a-5832-4149-9fe0-d17906c75884), S3 Extended Request ID: 2dfe849a-5832-4149-9fe0-d17906c75884
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1742)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1371)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1347)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1127)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:784)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:752)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5052)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4998)
	at com.amazonaws.services.s3.AmazonS3Client.createBucket(AmazonS3Client.java:1094)
	at io.minio.awssdk.tests.FunctionalTests.initTests(Unknown Source)
	at io.minio.awssdk.tests.FunctionalTests.main(Unknown Source)
(2/15) Running aws-sdk-php tests ... FAILED in 1 seconds
PHP Fatal error:  Uncaught exception 'Aws\S3\Exception\S3Exception' with message 'Error executing "DeleteObject" on "http://minio-c2.minio.io:30304/aws-sdk-php-5efd6aeb0d4f1/obj1"; AWS HTTP error: Server error: `DELETE http://minio-c2.minio.io:30304/aws-sdk-php-5efd6aeb0d4f1/obj1` resulted in a `503 Service Unavailable` response:
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, pl (truncated...)
 XMinioServerNotInitialized (server): Server not initialized, please try again. - <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key>obj1</Key><BucketName>aws-sdk-php-5efd6aeb0d4f1</BucketName><Resource>/aws-sdk-php-5efd6aeb0d4f1/obj1</Resource><RequestId>161DD70A827BFCBB</RequestId><HostId>2dfe849a-5832-4149-9fe0-d17906c75884</HostId></Error>'

GuzzleHttp\Exception\ServerException: Server error: `PUT http://minio-c2.minio.io:30304/aws-sdk-php-5efd6aeb0d4f1` resulted in  in /mint/run/core/aws-sdk-php/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php on line 195
(2/15) Running aws-sdk-ruby tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2130.68,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(2/15) Running awscli tests ... FAILED in 12 seconds
{
  "name": "awscli",
  "duration": 9641,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(2/15) Running healthcheck tests ... done in 1 seconds
(3/15) Running mc tests ... FAILED in 0 seconds
(3/15) Running minio-dotnet tests ... done in 32 seconds
(4/15) Running minio-go tests ... FAILED in 1 minutes and 15 seconds
{
  "args": {
    "bucketName": "minio-go-test-bsmlq3gltwp4ddyi",
    "region": "eu-west-1"
  },
  "duration": 75402,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(4/15) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "makeBucket(MakeBucketArgs args)",
  "duration": 394,
  "status": "FAIL",
  "error": "java.lang.IllegalArgumentException: unknown error code string 'XMinioServerNotInitialized' >>> [io.minio.Xml.unmarshal(Xml.java:68), io.minio.MinioClient.execute(MinioClient.java:1050), io.minio.MinioClient.makeBucket(MinioClient.java:3773), FunctionalTest.makeBucket_test1(FunctionalTest.java:435), FunctionalTest.runTests(FunctionalTest.java:3772), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(4/15) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(4/15) Running minio-py tests ... FAILED in 10 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "function": "make_bucket(bucket_name, location)",
  "args": {
    "bucket_name": "minio-py-test-4ba5b591-723c-4fad-9936-40ceb27eee10",
    "location": "default value ('us-east-1')"
  },
  "duration": 600419,
  "message": "HTTPConnectionPool(host='minio-c2.minio.io', port=30304): Max retries exceeded with url: /minio-py-test-4ba5b591-723c-4fad-9936-40ceb27eee10/ (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 188, in test_make_bucket_default_region\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 371, in make_bucket\n    headers=headers)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 336, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 805, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c2.minio.io', port=30304): Max retries exceeded with url: /minio-py-test-4ba5b591-723c-4fad-9936-40ceb27eee10/ (Caused by ResponseError('too many 503 error responses',))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1968, in main\n    test_make_bucket_default_region(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 196, in test_make_bucket_default_region\n    raise Exception(err)\nException: HTTPConnectionPool(host='minio-c2.minio.io', port=30304): Max retries exceeded with url: /minio-py-test-4ba5b591-723c-4fad-9936-40ceb27eee10/ (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(4/15) Running s3cmd tests ... FAILED in 46 seconds
{
  "name": "s3cmd",
  "duration": "45377",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(4/15) Running s3select tests ... FAILED in 1 hours 51 minutes and 6 seconds
{
  "name": "s3select:test_sql_operators",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "total_tests": 20,
    "total_success": 8
  },
  "duration": 6613745,
  "message": "HTTPConnectionPool(host='minio-c2.minio.io', port=30304): Max retries exceeded with url: /s3select-test-14012f58-ee81-4f40-9076-68ed649623ca?location= (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 51, in main\n    test_sql_operators(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 131, in test_sql_operators\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 129, in test_sql_operators\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 89, in test_sql_expressions\n    input_serialization, output_serialization, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 74, in test_sql_expressions_custom_input_output\n    client.remove_object(bucket_name, object_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1171, in remove_object\n    object_name=object_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1984, in _url_open\n    region = self._get_bucket_region(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1919, in _get_bucket_region\n    region = self._get_bucket_location(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1950, in _get_bucket_location\n    headers=headers)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 336, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 805, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c2.minio.io', port=30304): Max retries exceeded with url: /s3select-test-14012f58-ee81-4f40-9076-68ed649623ca?location= (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(4/15) Running security tests ... done in 0 seconds

Executed 4 out of 15 tests successfully.
```
<a name="ab872c68-9359-4b26-96d8-4ebc96741930"></a>
### 9957-a4426e9/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31115
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 61e8657f95ae:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 3 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 33 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... done in 2 minutes and 37 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 5 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 44 seconds
(9/15) Running minio-go tests ... done in 6 minutes and 36 seconds
(10/15) Running minio-java tests ... FAILED in 5 minutes and 25 seconds
{
  "name": "minio-java",
  "function": "removeObjects(RemoveObjectsArgs args)",
  "duration": 1220,
  "status": "FAIL",
  "error": "org.simpleframework.xml.core.ValueRequiredException: Unable to satisfy @org.simpleframework.xml.Element(name=\"Key\", type=void.class, data=false, required=true) on field 'name' private java.lang.String io.minio.messages.DeletedObject.name for class io.minio.messages.DeletedObject at line 2 >>> [io.minio.Xml.unmarshal(Xml.java:68), io.minio.MinioClient.deleteObjects(MinioClient.java:7354), io.minio.MinioClient$1$1.populate(MinioClient.java:2945), io.minio.MinioClient$1$1.hasNext(MinioClient.java:2977), FunctionalTest.removeObjects(FunctionalTest.java:1395), FunctionalTest.removeObjects_test1(FunctionalTest.java:1534), FunctionalTest.runTests(FunctionalTest.java:3812), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(10/15) Running minio-js tests ... done in 2 minutes and 39 seconds
(11/15) Running minio-py tests ... done in 9 minutes and 40 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 27 seconds
(13/15) Running s3select tests ... done in 1 minutes and 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="5353b93a-b41b-4628-bc8c-4bcf772e8900"></a>
### 9957-a4426e9/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:32763
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 50151074abcc:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 3 seconds
(3/15) Running aws-sdk-php tests ... FAILED in 2 seconds
{
  "name": "aws-sdk-php",
  "function": "uploadPart ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5efd69bb1ce87",
    "Object": "obj1"
  },
  "duration": "393",
  "status": "FAIL",
  "error": "Error executing \"CompleteMultipartUpload\" on \"http://minio-dev3.minio.io:32763/aws-sdk-php-5efd69bb1ce87/obj1?uploadId=non-existent\"; AWS HTTP error: Client error: `POST http://minio-dev3.minio.io:32763/aws-sdk-php-5efd69bb1ce87/obj1?uploadId=non-existent` resulted in a `404 Not Found` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message> (truncated...)\n NoSuchKey (client): The specified key does not exist. - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>obj1</Key><BucketName>aws-sdk-php-5efd69bb1ce87</BucketName><Resource>/aws-sdk-php-5efd69bb1ce87/obj1</Resource><RequestId>161DD6C44E2DD112</RequestId><HostId>e4fafa85-745d-4413-b4d7-470013f2d8b5</HostId></Error>"
}
(3/15) Running aws-sdk-ruby tests ... done in 5 seconds
(4/15) Running awscli tests ... done in 2 minutes and 7 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 25 seconds
(7/15) Running minio-dotnet tests ... done in 53 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 13 seconds
{
  "args": {
    "bucketName": "minio-go-test-onqmil6jgsy404md",
    "objectName": "f6dgmaeph6ldilj94s5hnnmmroe0en",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 3203,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "RemoveObjectsWithOptions(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjectsWithOptions",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 19 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[SSE-S3]",
  "duration": 27,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-17t99di, objectName = minio-java-test-29iqe1v, resource = /minio-java-test-17t99di/minio-java-test-29iqe1v, requestId = 161DD70A78E3A360, hostId = e4fafa85-745d-4413-b4d7-470013f2d8b5)\nrequest={method=PUT, url=http://minio-dev3.minio.io:32763/minio-java-test-17t99di/minio-java-test-29iqe1v, headers=X-Amz-Server-Side-Encryption: AES256\nContent-Type: application/javascript\nHost: minio-dev3.minio.io:32763\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/7.1.0\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20200702T050443Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200702/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-server-side-encryption, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 402\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-07-02T04-53-42Z\nVary: Origin\nX-Amz-Request-Id: 161DD70A78E3A360\nX-Xss-Protection: 1; mode=block\nDate: Thu, 02 Jul 2020 05:04:43 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1135), io.minio.MinioClient.putObject(MinioClient.java:7582), io.minio.MinioClient.putObject(MinioClient.java:4682), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.testPutObject(FunctionalTest.java:740), FunctionalTest.putObject_test(FunctionalTest.java:887), FunctionalTest.runTests(FunctionalTest.java:3789), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(8/15) Running minio-js tests ... done in 51 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 17 seconds
(10/15) Running s3cmd tests ... done in 19 seconds
(11/15) Running s3select tests ... done in 4 seconds
(12/15) Running security tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
<a name="a5b24040-fdff-4334-8d03-6e6e7524e3fe"></a>
### 9957-a4426e9/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:30756
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e0ecb95ec1dd:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 3 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 5 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 25 seconds
(8/15) Running minio-dotnet tests ... done in 30 seconds
(9/15) Running minio-go tests ... done in 30 seconds
(10/15) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[SSE-S3]",
  "duration": 12,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-50kfr0, objectName = minio-java-test-1gj4723, resource = /minio-java-test-50kfr0/minio-java-test-1gj4723, requestId = 161DD6F1635736C2, hostId = bf3a1643-ea05-4de8-b60e-31c913771b52)\nrequest={method=PUT, url=http://minio-dev4.minio.io:30756/minio-java-test-50kfr0/minio-java-test-1gj4723, headers=X-Amz-Server-Side-Encryption: AES256\nContent-Type: application/javascript\nHost: minio-dev4.minio.io:30756\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/7.1.0\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20200702T050255Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200702/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-server-side-encryption, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 400\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-07-02T04-53-42Z\nVary: Origin\nX-Amz-Request-Id: 161DD6F1635736C2\nX-Xss-Protection: 1; mode=block\nDate: Thu, 02 Jul 2020 05:02:55 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1135), io.minio.MinioClient.putObject(MinioClient.java:7582), io.minio.MinioClient.putObject(MinioClient.java:4682), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.testPutObject(FunctionalTest.java:740), FunctionalTest.putObject_test(FunctionalTest.java:887), FunctionalTest.runTests(FunctionalTest.java:3789), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(10/15) Running minio-js tests ... done in 34 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 31 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 3 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running security tests ... done in 1 seconds

Executed 14 out of 15 tests successfully.
```
<a name="5d3a8aaa-f619-4543-b9d6-bb747f44fda6"></a>
### 9957-a4426e9/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30395
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9780f330bb70:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 3 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 1 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 13 seconds
(8/15) Running minio-dotnet tests ... done in 28 seconds
(9/15) Running minio-go tests ... done in 23 seconds
(10/15) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[SSE-S3]",
  "duration": 8,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-68ehlq, objectName = minio-java-test-3nppgps, resource = /minio-java-test-68ehlq/minio-java-test-3nppgps, requestId = 161DD6F950B52D46, hostId = 9c0fadd8-d63b-4c12-b699-261aed7f1b86)\nrequest={method=PUT, url=http://minio-dev7.minio.io:30395/minio-java-test-68ehlq/minio-java-test-3nppgps, headers=X-Amz-Server-Side-Encryption: AES256\nContent-Type: application/javascript\nHost: minio-dev7.minio.io:30395\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/7.1.0\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20200702T050329Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200702/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-server-side-encryption, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 400\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-07-02T04-53-42Z\nVary: Origin\nX-Amz-Request-Id: 161DD6F950B52D46\nX-Xss-Protection: 1; mode=block\nDate: Thu, 02 Jul 2020 05:03:29 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1135), io.minio.MinioClient.putObject(MinioClient.java:7582), io.minio.MinioClient.putObject(MinioClient.java:4682), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.testPutObject(FunctionalTest.java:740), FunctionalTest.putObject_test(FunctionalTest.java:887), FunctionalTest.runTests(FunctionalTest.java:3789), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(10/15) Running minio-js tests ... done in 30 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 15 seconds
(12/15) Running s3cmd tests ... done in 15 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="7e4fe1ba-8efa-4c67-9d71-70c161838903"></a>
### 9957-a4426e9/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31038
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 949b5b4fa517:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 1 minutes and 17 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-j6xica4qtfwq0k",
    "objectName": "nmhlezlrvsnxmlg3rslmkcaegz4wio"
  },
  "duration": 12592,
  "error": "NoSuchKey: The specified key does not exist.\n\tstatus code: 404, request id: 161DD6D476B8F436, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go uploadPart API failed for",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 14 seconds
(2/15) Running aws-sdk-php tests ... FAILED in 1 minutes and 2 seconds
{
  "name": "aws-sdk-php",
  "function": "listMultipartUploads ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5efd6a11e95cb",
    "Object": "obj1"
  },
  "duration": "2145",
  "status": "FAIL",
  "error": "Error executing \"AbortMultipartUpload\" on \"http://minio-c2.minio.io:31038/aws-sdk-php-5efd6a11e95cb/obj1-copy?uploadId=553f494a-3135-4965-b488-158d59928665\"; AWS HTTP error: Client error: `DELETE http://minio-c2.minio.io:31038/aws-sdk-php-5efd6a11e95cb/obj1-copy?uploadId=553f494a-3135-4965-b488-158d59928665` resulted in a `404 Not Found` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message> (truncated...)\n NoSuchKey (client): The specified key does not exist. - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>obj1-copy</Key><BucketName>aws-sdk-php-5efd6a11e95cb</BucketName><Resource>/aws-sdk-php-5efd6a11e95cb/obj1-copy</Resource><RequestId>161DD6E624C16DF1</RequestId><HostId>6b718802-0d97-4a9b-9554-e3bc1d4588f2</HostId></Error>"
}
(2/15) Running aws-sdk-ruby tests ... done in 4 minutes and 31 seconds
(3/15) Running awscli tests ... FAILED in 1 minutes and 32 seconds
{
  "name": "awscli",
  "duration": 17783,
  "function": "aws --endpoint-url http://minio-c2.minio.io:31038 s3api upload-part --bucket awscli-mint-test-bucket-12864 --key awscli-mint-test-bucket-12864-object --body /mint/data/datafile-0-b --upload-id bbab8889-99af-48a0-b468-49594f3dcc3f --part-number 1\n",
  "status": "FAIL",
  "error": ""
}
(3/15) Running healthcheck tests ... done in 1 seconds
(4/15) Running mc tests ... FAILED in 1 minutes and 20 seconds
{
  "name": "mc",
  "duration": "20699",
  "function": "test_mirror_list_objects",
  "status": "FAIL",
  "error": "mirror and list differs"
}
(4/15) Running minio-dotnet tests ... done in 13 minutes and 56 seconds
(5/15) Running minio-go tests ... FAILED in 2 minutes and 21 seconds
{
  "args": {
    "destination": {},
    "sourceList": [
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ],
          "X-Amz-Copy-Source-Range": [
            "bytes=0-5242879"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-y9msln1kdiz5vrub/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "2b5c91ad399409ec9b409b66f5bb00fb"
          ]
        }
      }
    ]
  },
  "duration": 13492,
  "error": "The specified key does not exist.",
  "function": "ComposeObject(destination, sourceList)",
  "message": "ComposeObject failed",
  "name": "minio-go: testComposeMultipleSources",
  "status": "FAIL"
}
(5/15) Running minio-java tests ... FAILED in 2 minutes and 56 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[SSE-S3]",
  "duration": 14,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-3ds0t4f, objectName = minio-java-test-2h2aqoa, resource = /minio-java-test-3ds0t4f/minio-java-test-2h2aqoa, requestId = 161DD85A1F3A31E7, hostId = 6b718802-0d97-4a9b-9554-e3bc1d4588f2)\nrequest={method=PUT, url=http://minio-c2.minio.io:31038/minio-java-test-3ds0t4f/minio-java-test-2h2aqoa, headers=X-Amz-Server-Side-Encryption: AES256\nContent-Type: application/javascript\nHost: minio-c2.minio.io:31038\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/7.1.0\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20200702T052844Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200702/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-server-side-encryption, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 402\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-07-02T04-53-42Z\nVary: Origin\nX-Amz-Request-Id: 161DD85A1F3A31E7\nX-Xss-Protection: 1; mode=block\nDate: Thu, 02 Jul 2020 05:28:44 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1135), io.minio.MinioClient.putObject(MinioClient.java:7582), io.minio.MinioClient.putObject(MinioClient.java:4682), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.testPutObject(FunctionalTest.java:740), FunctionalTest.putObject_test(FunctionalTest.java:887), FunctionalTest.runTests(FunctionalTest.java:3789), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(5/15) Running minio-js tests ... FAILED in 1 minutes and 8 seconds
{
  "name": "minio-js",
  "function": "fPutObject(bucketName, objectName, filePath, metaData, callback)",
  "args": "bucketName:minio-js-test-b2d2cb81-1da6-4bfe-aaa2-2613772ff2fa, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB, metaData: [object Object]",
  "duration": 72,
  "status": "FAIL",
  "error": "S3Error: The specified key does not exist. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(5/15) Running minio-py tests ... FAILED in 57 seconds
{
  "name": "minio-py:test_fput_object_large_file",
  "function": "fput_object(bucket_name, object_name, file_path, content_type, metadata, sse, progress, part_size)",
  "args": {
    "bucket_name": "minio-py-test-1ce2c10a-a483-4333-844a-d8e03f593b96",
    "object_name": "076b1025-9a48-45ca-b91f-377660f02b32",
    "file_path": "/mint/data/datafile-11-MB",
    "metadata": {
      "x-amz-storage-class": "STANDARD_IA"
    }
  },
  "duration": 25752,
  "message": "NoSuchKey: message: The specified key does not exist.",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1762, in _stream_put_object\n    upload_result = pool.result()\n  File \"/usr/local/lib/python3.6/dist-packages/minio/thread_pool.py\", line 93, in result\n    raise self.exceptions_queue.get()\n  File \"/usr/local/lib/python3.6/dist-packages/minio/thread_pool.py\", line 57, in run\n    result = func(*args, **kargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1703, in _upload_part_routine\n    part_number, sse=sse, progress=progress)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1689, in _do_put_object\n    content_sha256=sha256_hex\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.NoSuchKey: NoSuchKey: message: The specified key does not exist.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 396, in test_fput_object_large_file\n    '-large', largefile, sse=sse)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 700, in fput_object\n    progress, part_size)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 961, in put_object\n    part_size=part_size)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1768, in _stream_put_object\n    upload_id)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1825, in _remove_incomplete_upload\n    headers={})\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.NoSuchKey: NoSuchKey: message: The specified key does not exist.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1993, in main\n    test_fput_object_large_file(client, largefile, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 400, in test_fput_object_large_file\n    raise Exception(err)\nException: NoSuchKey: message: The specified key does not exist.\n",
  "status": "FAIL"
}
(5/15) Running s3cmd tests ... FAILED in 30 seconds
{
  "name": "s3cmd",
  "duration": "1790",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "ERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-22985/s3cmd-test-object-16288 a9f9ba23-63ab-467c-ab31-672b1ba377c9\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id a9f9ba23-63ab-467c-ab31-672b1ba377c9 put ...\nto continue the upload.\nERROR: S3 error: 404 (NoSuchKey): The specified key does not exist."
}
(5/15) Running s3select tests ... done in 3 minutes and 17 seconds
(6/15) Running security tests ... done in 0 seconds

Executed 6 out of 15 tests successfully.
```
<a name="7857e461-3e83-4bd4-bb9c-940252e66f48"></a>
### 9957-a4426e9/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30932
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7019bd27216b:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 11 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... FAILED in 4 seconds
{
  "name": "aws-sdk-php",
  "function": "uploadPart ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5efd69c416b4f",
    "Object": "obj1"
  },
  "duration": "540",
  "status": "FAIL",
  "error": "Error executing \"CompleteMultipartUpload\" on \"http://minio-dev7.minio.io:30932/aws-sdk-php-5efd69c416b4f/obj1?uploadId=non-existent\"; AWS HTTP error: Client error: `POST http://minio-dev7.minio.io:30932/aws-sdk-php-5efd69c416b4f/obj1?uploadId=non-existent` resulted in a `404 Not Found` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message> (truncated...)\n NoSuchKey (client): The specified key does not exist. - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>obj1</Key><BucketName>aws-sdk-php-5efd69c416b4f</BucketName><Resource>/aws-sdk-php-5efd69c416b4f/obj1</Resource><RequestId>161DD6C67E1C2639</RequestId><HostId>47acdfcd-3a71-4d1b-b99e-076a02c8e655</HostId></Error>"
}
(3/15) Running aws-sdk-ruby tests ... done in 7 seconds
(4/15) Running awscli tests ... done in 2 minutes and 13 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 1 minutes and 22 seconds
(7/15) Running minio-dotnet tests ... done in 2 minutes and 2 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-h0owa1kd5utxiirk",
    "objectName": "694vjifanidravk1cm9ntiici292pg",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 3980,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "RemoveObjectsWithOptions(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjectsWithOptions",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 19 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[SSE-S3]",
  "duration": 13,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-3s278fs, objectName = minio-java-test-1atb6bh, resource = /minio-java-test-3s278fs/minio-java-test-1atb6bh, requestId = 161DD737A18291F9, hostId = 47acdfcd-3a71-4d1b-b99e-076a02c8e655)\nrequest={method=PUT, url=http://minio-dev7.minio.io:30932/minio-java-test-3s278fs/minio-java-test-1atb6bh, headers=X-Amz-Server-Side-Encryption: AES256\nContent-Type: application/javascript\nHost: minio-dev7.minio.io:30932\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/7.1.0\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20200702T050757Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200702/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-server-side-encryption, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 402\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-07-02T04-53-42Z\nVary: Origin\nX-Amz-Request-Id: 161DD737A18291F9\nX-Xss-Protection: 1; mode=block\nDate: Thu, 02 Jul 2020 05:07:57 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1135), io.minio.MinioClient.putObject(MinioClient.java:7582), io.minio.MinioClient.putObject(MinioClient.java:4682), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.testPutObject(FunctionalTest.java:740), FunctionalTest.putObject_test(FunctionalTest.java:887), FunctionalTest.runTests(FunctionalTest.java:3789), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(8/15) Running minio-js tests ... done in 58 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 56 seconds
(10/15) Running s3cmd tests ... done in 23 seconds
(11/15) Running s3select tests ... done in 7 seconds
(12/15) Running security tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
<a name="d5f4b67b-7edc-4ef4-bfe3-75375a88650a"></a>
### 9957-a4426e9/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:30525
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b1cf79e85991:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 4 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 24 seconds
(8/15) Running minio-dotnet tests ... done in 27 seconds
(9/15) Running minio-go tests ... done in 27 seconds
(10/15) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[SSE-S3]",
  "duration": 12,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-22bghig, objectName = minio-java-test-2eojrfm, resource = /minio-java-test-22bghig/minio-java-test-2eojrfm, requestId = 161DD6EF13851AB2, hostId = 8f5ff140-be0b-4de8-9bd7-a98210627aa0)\nrequest={method=PUT, url=http://minio-dev4.minio.io:30525/minio-java-test-22bghig/minio-java-test-2eojrfm, headers=X-Amz-Server-Side-Encryption: AES256\nContent-Type: application/javascript\nHost: minio-dev4.minio.io:30525\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/7.1.0\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20200702T050245Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200702/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-server-side-encryption, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 402\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-07-02T04-53-42Z\nVary: Origin\nX-Amz-Request-Id: 161DD6EF13851AB2\nX-Xss-Protection: 1; mode=block\nDate: Thu, 02 Jul 2020 05:02:45 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1135), io.minio.MinioClient.putObject(MinioClient.java:7582), io.minio.MinioClient.putObject(MinioClient.java:4682), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.testPutObject(FunctionalTest.java:740), FunctionalTest.putObject_test(FunctionalTest.java:887), FunctionalTest.runTests(FunctionalTest.java:3789), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(10/15) Running minio-js tests ... done in 44 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 17 seconds
(12/15) Running s3cmd tests ... done in 15 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/admin-heal-ops.go
+++ b/cmd/admin-heal-ops.go
@@ -629,7 +629,7 @@ func (h *healSequence) queueHealTask(source healSource, healType madmin.HealItem
 			// Object might have been deleted, by the time heal
 			// was attempted, we should ignore this object and
 			// return success.
-			if isErrObjectNotFound(res.err) {
+			if isErrObjectNotFound(res.err) || isErrVersionNotFound(res.err) {
 				return nil
 			}
 
@@ -657,7 +657,7 @@ func (h *healSequence) queueHealTask(source healSource, healType madmin.HealItem
 		if res.err != nil {
 			// Object might have been deleted, by the time heal
 			// was attempted, we should ignore this object and return success.
-			if isErrObjectNotFound(res.err) {
+			if isErrObjectNotFound(res.err) || isErrVersionNotFound(res.err) {
 				return nil
 			}
 			// Only report object error
@@ -769,7 +769,7 @@ func (h *healSequence) healMinioSysMeta(metaPrefix string) func() error {
 			}, madmin.HealItemBucketMetadata)
 			// Object might have been deleted, by the time heal
 			// was attempted we ignore this object an move on.
-			if isErrObjectNotFound(herr) {
+			if isErrObjectNotFound(herr) || isErrVersionNotFound(herr) {
 				return nil
 			}
 			return herr

--- a/cmd/erasure-multipart.go
+++ b/cmd/erasure-multipart.go
@@ -41,7 +41,7 @@ func (er erasureObjects) getMultipartSHADir(bucket, object string) string {
 
 // checkUploadIDExists - verify if a given uploadID exists and is valid.
 func (er erasureObjects) checkUploadIDExists(ctx context.Context, bucket, object, uploadID string) error {
-	_, err := er.getObjectInfo(ctx, minioMetaMultipartBucket, er.getUploadIDDir(bucket, object, uploadID), ObjectOptions{})
+	_, _, _, err := er.getObjectFileInfo(ctx, minioMetaMultipartBucket, er.getUploadIDDir(bucket, object, uploadID), ObjectOptions{})
 	return err
 }
 

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -145,7 +145,20 @@ func (er erasureObjects) GetObjectNInfo(ctx context.Context, bucket, object stri
 		return nil, toObjectErr(err, bucket, object)
 	}
 
-	fn, off, length, nErr := NewGetObjectReader(rs, fi.ToObjectInfo(bucket, object), opts)
+	objInfo := fi.ToObjectInfo(bucket, object)
+	if objInfo.DeleteMarker {
+		if opts.VersionID == "" {
+			return &GetObjectReader{
+				ObjInfo: objInfo,
+			}, toObjectErr(errFileNotFound, bucket, object)
+		}
+		// Make sure to return object info to provide extra information.
+		return &GetObjectReader{
+			ObjInfo: objInfo,
+		}, toObjectErr(errMethodNotAllowed, bucket, object)
+	}
+
+	fn, off, length, nErr := NewGetObjectReader(rs, objInfo, opts)
 	if nErr != nil {
 		return nil, nErr
 	}
@@ -310,6 +323,14 @@ func (er erasureObjects) getObject(ctx context.Context, bucket, object string, s
 	if err != nil {
 		return toObjectErr(err, bucket, object)
 	}
+	if fi.Deleted {
+		if opts.VersionID == "" {
+			return toObjectErr(errFileNotFound, bucket, object)
+		}
+		// Make sure to return object info to provide extra information.
+		return toObjectErr(errMethodNotAllowed, bucket, object)
+	}
+
 	return er.getObjectWithFileInfo(ctx, bucket, object, startOffset, length, writer, etag, opts, fi, metaArr, onlineDisks)
 }
 
@@ -358,12 +379,7 @@ func (er erasureObjects) GetObjectInfo(ctx context.Context, bucket, object strin
 		return info, nil
 	}
 
-	info, err = er.getObjectInfo(ctx, bucket, object, opts)
-	if err != nil {
-		return info, toObjectErr(err, bucket, object)
-	}
-
-	return info, nil
+	return er.getObjectInfo(ctx, bucket, object, opts)
 }
 
 func (er erasureObjects) getObjectFileInfo(ctx context.Context, bucket, object string, opts ObjectOptions) (fi FileInfo, metaArr []FileInfo, onlineDisks []StorageAPI, err error) {
@@ -397,7 +413,7 @@ func (er erasureObjects) getObjectFileInfo(ctx context.Context, bucket, object s
 func (er erasureObjects) getObjectInfo(ctx context.Context, bucket, object string, opts ObjectOptions) (objInfo ObjectInfo, err error) {
 	fi, _, _, err := er.getObjectFileInfo(ctx, bucket, object, opts)
 	if err != nil {
-		return objInfo, err
+		return objInfo, toObjectErr(err, bucket, object)
 	}
 
 	if fi.Deleted {

--- a/cmd/erasure-zones.go
+++ b/cmd/erasure-zones.go
@@ -465,19 +465,22 @@ func (z *erasureZones) GetObjectNInfo(ctx context.Context, bucket, object string
 	}
 
 	for _, zone := range z.zones {
-		gr, err := zone.GetObjectNInfo(ctx, bucket, object, rs, h, lockType, opts)
+		gr, err = zone.GetObjectNInfo(ctx, bucket, object, rs, h, lockType, opts)
 		if err != nil {
-			if isErrObjectNotFound(err) {
+			if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
 				continue
 			}
 			nsUnlocker()
-			return nil, err
+			return gr, err
 		}
 		gr.cleanUpFns = append(gr.cleanUpFns, nsUnlocker)
 		return gr, nil
 	}
 	nsUnlocker()
-	return nil, ObjectNotFound{Bucket: bucket, Object: object}
+	if opts.VersionID != "" {
+		return gr, VersionNotFound{Bucket: bucket, Object: object, VersionID: opts.VersionID}
+	}
+	return gr, ObjectNotFound{Bucket: bucket, Object: object}
 }
 
 func (z *erasureZones) GetObject(ctx context.Context, bucket, object string, startOffset int64, length int64, writer io.Writer, etag string, opts ObjectOptions) error {
@@ -491,9 +494,10 @@ func (z *erasureZones) GetObject(ctx context.Context, bucket, object string, sta
 	if z.SingleZone() {
 		return z.zones[0].GetObject(ctx, bucket, object, startOffset, length, writer, etag, opts)
 	}
+
 	for _, zone := range z.zones {
 		if err := zone.GetObject(ctx, bucket, object, startOffset, length, writer, etag, opts); err != nil {
-			if isErrObjectNotFound(err) {
+			if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
 				continue
 			}
 			return err
@@ -503,7 +507,7 @@ func (z *erasureZones) GetObject(ctx context.Context, bucket, object string, sta
 	return ObjectNotFound{Bucket: bucket, Object: object}
 }
 
-func (z *erasureZones) GetObjectInfo(ctx context.Context, bucket, object string, opts ObjectOptions) (ObjectInfo, error) {
+func (z *erasureZones) GetObjectInfo(ctx context.Context, bucket, object string, opts ObjectOptions) (objInfo ObjectInfo, err error) {
 	// Lock the object before reading.
 	lk := z.NewNSLock(ctx, bucket, object)
 	if err := lk.GetRLock(globalObjectTimeout); err != nil {
@@ -515,16 +519,19 @@ func (z *erasureZones) GetObjectInfo(ctx context.Context, bucket, object string,
 		return z.zones[0].GetObjectInfo(ctx, bucket, object, opts)
 	}
 	for _, zone := range z.zones {
-		objInfo, err := zone.GetObjectInfo(ctx, bucket, object, opts)
+		objInfo, err = zone.GetObjectInfo(ctx, bucket, object, opts)
 		if err != nil {
-			if isErrObjectNotFound(err) {
+			if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
 				continue
 			}
 			return objInfo, err
 		}
 		return objInfo, nil
 	}
-	return ObjectInfo{}, ObjectNotFound{Bucket: bucket, Object: object}
+	if opts.VersionID != "" {
+		return objInfo, VersionNotFound{Bucket: bucket, Object: object, VersionID: opts.VersionID}
+	}
+	return objInfo, ObjectNotFound{Bucket: bucket, Object: object}
 }
 
 // PutObject - writes an object to least used erasure zone.
@@ -565,7 +572,7 @@ func (z *erasureZones) DeleteObject(ctx context.Context, bucket string, object s
 		if err == nil {
 			return objInfo, nil
 		}
-		if err != nil && !isErrObjectNotFound(err) {
+		if err != nil && !isErrObjectNotFound(err) && !isErrVersionNotFound(err) {
 			break
 		}
 	}
@@ -595,7 +602,7 @@ func (z *erasureZones) DeleteObjects(ctx context.Context, bucket string, objects
 		deletedObjects, errs := zone.DeleteObjects(ctx, bucket, objects, opts)
 		for i, derr := range errs {
 			if derrs[i] == nil {
-				if derr != nil && !isErrObjectNotFound(derr) {
+				if derr != nil && !isErrObjectNotFound(derr) && !isErrVersionNotFound(derr) {
 					derrs[i] = derr
 				}
 			}
@@ -1918,7 +1925,7 @@ func (z *erasureZones) HealObject(ctx context.Context, bucket, object, versionID
 	for _, zone := range z.zones {
 		result, err := zone.HealObject(ctx, bucket, object, versionID, opts)
 		if err != nil {
-			if isErrObjectNotFound(err) {
+			if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
 				continue
 			}
 			return result, err
@@ -2022,7 +2029,7 @@ func (z *erasureZones) PutObjectTags(ctx context.Context, bucket, object string,
 	for _, zone := range z.zones {
 		err := zone.PutObjectTags(ctx, bucket, object, tags, opts)
 		if err != nil {
-			if isErrObjectNotFound(err) {
+			if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
 				continue
 			}
 			return err
@@ -2050,7 +2057,7 @@ func (z *erasureZones) DeleteObjectTags(ctx context.Context, bucket, object stri
 	for _, zone := range z.zones {
 		err := zone.DeleteObjectTags(ctx, bucket, object, opts)
 		if err != nil {
-			if isErrObjectNotFound(err) {
+			if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
 				continue
 			}
 			return err
@@ -2078,7 +2085,7 @@ func (z *erasureZones) GetObjectTags(ctx context.Context, bucket, object string,
 	for _, zone := range z.zones {
 		tags, err := zone.GetObjectTags(ctx, bucket, object, opts)
 		if err != nil {
-			if isErrObjectNotFound(err) {
+			if isErrObjectNotFound(err) || isErrVersionNotFound(err) {
 				continue
 			}
 			return tags, err

--- a/cmd/generic-handlers.go
+++ b/cmd/generic-handlers.go
@@ -214,7 +214,8 @@ func guessIsHealthCheckReq(req *http.Request) bool {
 	aType := getRequestAuthType(req)
 	return aType == authTypeAnonymous && (req.Method == http.MethodGet || req.Method == http.MethodHead) &&
 		(req.URL.Path == healthCheckPathPrefix+healthCheckLivenessPath ||
-			req.URL.Path == healthCheckPathPrefix+healthCheckReadinessPath)
+			req.URL.Path == healthCheckPathPrefix+healthCheckReadinessPath ||
+			req.URL.Path == healthCheckPathPrefix+healthCheckClusterPath)
 }
 
 // guessIsMetricsReq - returns true if incoming request looks

--- a/cmd/object-api-errors.go
+++ b/cmd/object-api-errors.go
@@ -519,6 +519,12 @@ func isErrObjectNotFound(err error) bool {
 	return errors.As(err, &objNotFound)
 }
 
+// isErrVersionNotFound - Check if error type is VersionNotFound.
+func isErrVersionNotFound(err error) bool {
+	var versionNotFound VersionNotFound
+	return errors.As(err, &versionNotFound)
+}
+
 // PreConditionFailed - Check if copy precondition failed
 type PreConditionFailed struct{}
 

--- a/cmd/object-handlers.go
+++ b/cmd/object-handlers.go
@@ -116,7 +116,7 @@ func (api objectAPIHandlers) SelectObjectContentHandler(w http.ResponseWriter, r
 	// get gateway encryption options
 	opts, err := getOpts(ctx, r, bucket, object)
 	if err != nil {
-		writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
+		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
 	}
 
@@ -192,6 +192,14 @@ func (api objectAPIHandlers) SelectObjectContentHandler(w http.ResponseWriter, r
 
 	objInfo, err := getObjectInfo(ctx, bucket, object, opts)
 	if err != nil {
+		if globalBucketVersioningSys.Enabled(bucket) {
+			// Versioning enabled quite possibly object is deleted might be delete-marker
+			// if present set the headers, no idea why AWS S3 sets these headers.
+			if objInfo.VersionID != "" && objInfo.DeleteMarker {
+				w.Header()[xhttp.AmzVersionID] = []string{objInfo.VersionID}
+				w.Header()[xhttp.AmzDeleteMarker] = []string{strconv.FormatBool(objInfo.DeleteMarker)}
+			}
+		}
 		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
 	}
@@ -256,7 +264,7 @@ func (api objectAPIHandlers) SelectObjectContentHandler(w http.ResponseWriter, r
 			case crypto.SSEC.IsEncrypted(objInfo.UserDefined):
 				// Validate the SSE-C Key set in the header.
 				if _, err = crypto.SSEC.UnsealObjectKey(r.Header, objInfo.UserDefined, bucket, object); err != nil {
-					writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
+					writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 					return
 				}
 				w.Header().Set(crypto.SSECAlgorithm, r.Header.Get(crypto.SSECAlgorithm))
@@ -313,7 +321,7 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 	// get gateway encryption options
 	opts, err := getOpts(ctx, r, bucket, object)
 	if err != nil {
-		writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
+		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
 	}
 
@@ -379,6 +387,14 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 
 	gr, err := getObjectNInfo(ctx, bucket, object, rs, r.Header, readLock, opts)
 	if err != nil {
+		if globalBucketVersioningSys.Enabled(bucket) && gr != nil {
+			// Versioning enabled quite possibly object is deleted might be delete-marker
+			// if present set the headers, no idea why AWS S3 sets these headers.
+			if gr.ObjInfo.VersionID != "" && gr.ObjInfo.DeleteMarker {
+				w.Header()[xhttp.AmzVersionID] = []string{gr.ObjInfo.VersionID}
+				w.Header()[xhttp.AmzDeleteMarker] = []string{strconv.FormatBool(gr.ObjInfo.DeleteMarker)}
+			}
+		}
 		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
 	}
@@ -435,6 +451,7 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 		statusCodeWritten = true
 		w.WriteHeader(http.StatusPartialContent)
 	}
+
 	// Write object content to response body
 	if _, err = io.Copy(httpWriter, gr); err != nil {
 		if !httpWriter.HasWritten() && !statusCodeWritten { // write error response only if no data or headers has been written to client yet
@@ -552,6 +569,14 @@ func (api objectAPIHandlers) HeadObjectHandler(w http.ResponseWriter, r *http.Re
 
 	objInfo, err := getObjectInfo(ctx, bucket, object, opts)
 	if err != nil {
+		if globalBucketVersioningSys.Enabled(bucket) {
+			// Versioning enabled quite possibly object is deleted might be delete-marker
+			// if present set the headers, no idea why AWS S3 sets these headers.
+			if objInfo.VersionID != "" && objInfo.DeleteMarker {
+				w.Header()[xhttp.AmzVersionID] = []string{objInfo.VersionID}
+				w.Header()[xhttp.AmzDeleteMarker] = []string{strconv.FormatBool(objInfo.DeleteMarker)}
+			}
+		}
 		writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
 		return
 	}
@@ -860,6 +885,14 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 	if err != nil {
 		if isErrPreconditionFailed(err) {
 			return
+		}
+		if globalBucketVersioningSys.Enabled(srcBucket) && gr != nil {
+			// Versioning enabled quite possibly object is deleted might be delete-marker
+			// if present set the headers, no idea why AWS S3 sets these headers.
+			if gr.ObjInfo.VersionID != "" && gr.ObjInfo.DeleteMarker {
+				w.Header()[xhttp.AmzVersionID] = []string{gr.ObjInfo.VersionID}
+				w.Header()[xhttp.AmzDeleteMarker] = []string{strconv.FormatBool(gr.ObjInfo.DeleteMarker)}
+			}
 		}
 		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
@@ -1408,7 +1441,7 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 	var opts ObjectOptions
 	opts, err = putOpts(ctx, r, bucket, object, metadata)
 	if err != nil {
-		writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
+		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
 	}
 
@@ -1620,7 +1653,7 @@ func (api objectAPIHandlers) NewMultipartUploadHandler(w http.ResponseWriter, r 
 
 	opts, err := putOpts(ctx, r, bucket, object, metadata)
 	if err != nil {
-		writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
+		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
 	}
 	newMultipartUpload := objectAPI.NewMultipartUpload
@@ -1770,6 +1803,14 @@ func (api objectAPIHandlers) CopyObjectPartHandler(w http.ResponseWriter, r *htt
 	if err != nil {
 		if isErrPreconditionFailed(err) {
 			return
+		}
+		if globalBucketVersioningSys.Enabled(srcBucket) && gr != nil {
+			// Versioning enabled quite possibly object is deleted might be delete-marker
+			// if present set the headers, no idea why AWS S3 sets these headers.
+			if gr.ObjInfo.VersionID != "" && gr.ObjInfo.DeleteMarker {
+				w.Header()[xhttp.AmzVersionID] = []string{gr.ObjInfo.VersionID}
+				w.Header()[xhttp.AmzDeleteMarker] = []string{strconv.FormatBool(gr.ObjInfo.DeleteMarker)}
+			}
 		}
 		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return

--- a/cmd/web-handlers.go
+++ b/cmd/web-handlers.go
@@ -756,7 +756,7 @@ next:
 		}
 	}
 
-	if err != nil && !isErrObjectNotFound(err) {
+	if err != nil && !isErrObjectNotFound(err) && !isErrVersionNotFound(err) {
 		// Ignore object not found error.
 		return toJSONError(ctx, err, args.BucketName, "")
 	}

